### PR TITLE
[New Devices] Cisco IOL and IOL L2 device support with containerlab provider.

### PR DESCRIPTION
**Pull request by DanPartelly**

### CODE REVIEW

**Comment by ipspace** [comment]
I like that you used **iol** as a parent of **ioll2**, but it would be even better to have **iosv** as parent of **iol** and then only set the differences in `iol.yml`

The documentation will have to be fixed in numerous other places (if IOL is IOS/XE then every time IOS/XE is mentioned together with a list of devices that implies), but I can do that afterwards.

Finally, it would be nice to have a `caveats.md` section for Cisco IOL and Cisco IOLL2 and list device-specific caveats (like containerlab release and vrnetlabs fork) there.

File: netsim/ansible/templates/initial/ios.j2 [hunk 1]
```diff
@@ -5,7 +5,7 @@ no ip domain lookup
 lldp run
 !
 {% for hname,hdata in hosts.items() if hname != inventory_hostname %}
-{%   set ipv6_addr = hdata.ipv6|default([]) if netlab_device_type != 'iosvl2' else [] %}
+{%   set ipv6_addr = hdata.ipv6|default([]) if netlab_device_type != 'iosvl2' or netlab_device_type != 'ioll2' else [] %}
```
Comment (ipspace) [inline netsim/ansible/templates/initial/ios.j2:8]: `netlab_device_type not in ['iosvl2','ioll2']` would be more concise

File: netsim/ansible/templates/initial/ios.j2 [hunk 2]
```diff
@@ -48,7 +48,7 @@ interface {{ l.ifname }}
  no shutdown
 {% if l.type == 'vlan_member' and l.vlan.access_id is defined %}
  encapsulation dot1Q {{ l.vlan.access_id }}
-{% elif l.virtual_interface is not defined and netlab_device_type == 'iosvl2' %}
+{% elif l.virtual_interface is not defined and (netlab_device_type == 'iosvl2' or netlab_device_type == 'ioll2') %}
```
Comment (ipspace) [inline netsim/ansible/templates/initial/ios.j2:51]: Same as above, `netlab_device_type in ['iosvl2','ioll2']`

File: netsim/devices/iol.yml [hunk 3]
```diff
@@ -0,0 +1,100 @@
+---
+description: Cisco IOL
+interface_name: Ethernet{ifindex // 4}/{ifindex % 4}
```
Comment (ipspace) [inline netsim/devices/iol.yml:3]: Is this correct? The only other platform using EthernetX/Y naming scheme is IOSv

**Comment by ipspace** [comment]
Apart from a few minor details, great job 🎉

File: netsim/devices/iol.yml (hunk 3 above)
Comment (DanPartelly) [reply netsim/devices/iol.yml:3]: show ip int br

Ethernet0/0            192.168.121.101 YES TFTP   up                    up      
Ethernet0/1            10.1.0.1        YES manual up                    up      
Ethernet0/2            10.1.0.5        YES manual up                    up      
Ethernet0/3            10.1.0.9        YES manual up                    up      
Ethernet1/0            10.1.0.13       YES manual up                    up      
Ethernet1/1            10.1.0.17       YES manual up                    up      
Ethernet1/2            10.1.0.21       YES manual up                    up      
Ethernet1/3            10.1.0.25       YES manual up                    up      
Ethernet2/0            10.1.0.29       YES manual up                    up      
Ethernet2/1            10.1.0.33       YES manual up                    up

File: docs/platforms.md [hunk 4]
```diff
@@ -18,6 +18,8 @@
 | Cisco ASAv                                | asav               |
 | Cisco Catalyst 8000v [❗](caveats-cat8000v) | cat8000v                |
 | Cisco CSR 1000v [❗](caveats-csr)         | csr                |
+| Cisco IOL                                 | iol                |
```
[the remark this answers is not in the document]
Comment (DanPartelly) [reply docs/platforms.md:21]: I will do this as well later this week.

**Comment by ipspace** [comment]
Added an image fix to IOL definition. Merging the PR, the documentation can be fixed later.

File: docs/platforms.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (DanPartelly) [reply docs/platforms.md:21]: Can you please be more specific on what you want to mention in docs where IOS/XE is previously mentioned ? 
The particular image with version 17.12.01  is  XE variant, so I guess we should coder this.

[the remark this answers is not in the document]
Comment (ipspace) [reply docs/platforms.md:21]: No worries, I'll fix that (I know where to look ;). So it's XE-based (like CSR and Cat8000v) and can support unnumbered IPv4? That's the major difference between IOSv and CSR IIRC.

**Pull request merged**